### DISCUSSION TITLE
开启codefirst 同步结构，在同步结构的before事件里判断实体是否需要同步，把不需要同的实体移除

### DISCUSSION THREAD

**Discussion by sqlite-me** [dotnetcore/FreeSql#794] [Q&A]
由于项目中使用了以前别人建的实体，实体结构与表结构不一致，在开发环境开启了自动同步结构，查询这些实体时会自动修改表结构，但是希望的是不修改。我想在同步结构的before事件里判断实体是否需要同步，把不需要同的实体移除。请问能不能做到，或者有没有其它方法！！

**Comment by luoyunchong** [comment]
1.全局关闭 **UseAutoSyncStructure(false)**   https://github.com/dotnetcore/FreeSql/wiki/%e5%85%a5%e9%97%a8#freesqlbuilder
2. 在实体类上加特性标签  **[Table(DisableSyncStructure = true)]**
 https://github.com/dotnetcore/FreeSql/wiki/%e5%ae%9e%e4%bd%93%e7%89%b9%e6%80%a7#%E7%A6%81%E7%94%A8%E8%BF%81%E7%A7%BB
3.FluentApi 设置某个表 https://github.com/dotnetcore/FreeSql/wiki/FluentApi
```
fsql.CodeFirst
                .ConfigEntity<Test2>(a =>
                {
                    a.DisableSyncStructure(true);
                });
```

**Reply by sqlite-me** [reply]
这些方法都不太理想，因为新旧项目混合使用，实体太多，并且不确定哪些实体会修改数据库表结构，希望在事件中拦截，并判断实体是否有Table特性来确定要不要同步结构，就更好

**Reply by luoyunchong** [reply]
先设置关闭 .全局关闭 UseAutoSyncStructure(false)
再看这个文档，通过反射或通过命名空间 获取Types[] 再调用 fsql.CodeFirst.SyncStructure(types);
https://github.com/dotnetcore/FreeSql/wiki/CodeFirst#%E6%89%B9%E9%87%8F%E7%94%9F%E6%88%90%E8%A1%A8%E7%BB%93%E6%9E%84
